lib/github_service/commands/run_test.rb

```diff
@@ -0,0 +1,302 @@
+      end
+
+      REPO_ONLY_REGEXP = /(?:[^\/]+\/)?(?<REPO_NAME_ONLY>[^@#]+)/.freeze
+      def repos_valid?(repos)
+        repos.all? do |repo|
+          name_only = repo.match(REPO_ONLY_REGEXP)[:REPO_NAME_ONLY]
+          GithubService.org_repos(issue.organization_name).include?(name_only)
+        end
+      end
+
+      ##### run_tests steps #####
+
+      # Clone repo (if needed) and initialize @rugged_repo
+      def clone_repo_if_necessary
+        repo_path = self.class.test_repo_clone_dir
+        if Dir.exist?(self.class.test_repo_clone_dir)
+          @rugged_repo = Rugged::Repository.new(repo_path)
+        else
+          url = self.class.test_repo_url
+          @rugged_repo = Rugged::Repository.clone_at(url, repo_path)
+        end
+        git_fetch
+      end
+
+      def create_cross_repo_test_branch
+        rugged_repo.create_branch(branch_name, "origin/master")
+        rugged_repo.checkout(branch_name)
+      end
+
+      def update_travis_yaml_content
+        Dir.chdir(@rugged_repo.workdir) do
+          content = YAML.load_file(".travis.yml")
+
+          content["env"] = {} unless content["env"]
+          content["env"]["global"] = ["REPOS=#{repos.join(',')}"]
+          content["env"]["matrix"] = test_repos.map { |repo| "TEST_REPO=#{repo}" }
+
+          File.write('.travis.yml', content.to_yaml)
+        end
+      end
+
+      def commit_travis_yaml_changes
+        index = rugged_repo.index
+        index.add('.travis.yml')
+        index.write
+
+        # rubocop:disable: Rails/TimeZone
+        bot       = self.class.bot_name
+        author    = { :name => issuer, :email => "no-name@example.com", :time => Time.now }
+        committer = { :name => bot,    :email => "#{bot}@manageiq.org", :time => Time.now }
+        # rubocop:enable: Rails/TimeZone
+
+        Rugged::Commit.create(
+          rugged_repo,
+          :author     => author,
+          :committer  => committer,
+          :parents    => [rugged_repo.last_commit].compact,
+          :tree       => index.write_tree(rugged_repo),
+          :update_ref => "HEAD",
+          :message    => <<~COMMIT_MSG
+            Running tests for #{issuer}
+
+            From Pull Request:  #{issue.fq_repo_name}##{issue.number}
+          COMMIT_MSG
+        )
+      end
+
+      def push_commit_to_remote
+        push_options = {}
+
+        if Settings.github_credentials.username && Settings.github_credentials.password
+          rugged_creds = Rugged::Credentials::UserPassword.new(
+            :username => Settings.github_credentials.username,
+            :password => Settings.github_credentials.password
+          )
+          push_options[:credentials] = rugged_creds
+        end
+
+        remote = @rugged_repo.remotes['origin']
+        remote.push(["refs/heads/#{branch_name}"], push_options)
+      end
+
+      def create_cross_repo_test_pull_request
+        fq_repo_name = "#{issue.organization_name}/#{File.basename(self.class.test_repo_url, '.*')}"
```

Comment:
Be aware that the test_repo_url may not actually be http based since it's just a remote and could be in any remote format...that is, in the miq-bot's deployment we'll likely use `git@github.com:ManageIQ/manageiq-cross_repo-tests` (which still works with your code, but I'm not sure you took that into account intentionally or not).

Reply:
We may want to move this line to a method like `test_repo_name` or something and have it live up near test_repo_url

Reply:
> Be aware that the test_repo_url may not actually be http based...

As you found out here:

https://github.com/ManageIQ/miq_bot/pull/454/files#r354919027

I was sort of making that assumption in general for this particular git repo case.  Not sure if I agree with past self looking back on this, but for now, I would say assume that I expect that `manageiq-cross_repo-tests` has to be in it's https form for this current implementation.

> We may want to move this line to a method like `test_repo_name` or something and have it live up near test_repo_url

Well, since `self.test_repo_name` already exists, maybe `fq_test_repo_name`?  Probably has to be an instance method, though, I guess we could just configure it from the `url` in the class as well.  🤔

Reply:
Yeah for first pass this is fine, but we'll likely have to fix that on second pass.

Reply:
`fq_test_repo_name` seems pretty good offhand

Reply:
Okay, so I am trying to solve for this right now, but it has proven to be a pain as currently there isn't an easy way to parse the `user@blah:org/repo.git` ssh form while using something like `URI` (at least that I am aware of).

I will mess with this some more, but since I am out next week, I wouldn't be able to address something like this terribly well, so am leaning more towards "just use HTTPS" for now, and we can address the above in a followup.  But that is just my 2 cents.

Reply:
For this PR, we are going to skip this.

I will open up an issue after this PR is merged to resolve this.  It is doable, just don't think it is worth solving for at the moment and supporting both.